### PR TITLE
Expose more of the flinkcluster CRD.

### PR DESCRIPTION
**Pull request by sabw8217**

### CODE REVIEW

File: charts/flink-job/templates/flinkjobcluster.yaml [hunk 1]
```diff
@@ -11,70 +11,72 @@ spec:
...
   envVars:
 {{ toYaml .Values.envVars | indent 4 }}
   serviceAccountName: {{ template "flink-job-cluster.fullname" . }}
   jobManager:
     accessScope: {{ .Values.jobManager.accessScope }}
     ports:
       ui: {{ .Values.jobManager.ports.ui }}
+      # TODO (Aaron/Albert) expose the other ports?
+{{- if .Values.jobManager.metrics.enabled }}
+    extraPorts:
```
Comment (sabw8217) [inline charts/flink-job/templates/flinkjobcluster.yaml:24]: I rearranged some stuff here to make it align with the flink-on-k8s-operator docs better

File: charts/flink-job/values.yaml [hunk 2]
```diff
@@ -127,12 +130,17 @@ job:
...
   # -- (List) Command-line args of the job
   args: ["--input", "./README.txt", "--output", "./OUTPUT.txt"]
-  # -- (`int`) Parallelism of the job
-  parallelism: 1
+  fromSavepoint:
   # -- (`int`) Automatically take a savepoint to the savepointsDir in this given interval
   autoSavepointSeconds: 30
   # -- (String) Directory to store automatically taken savepoints
   savepointsDir: /savepoint
+  allowNonRestoredState: false
```
Comment (diranged) [inline charts/flink-job/values.yaml:138]: Please add docstring

File: charts/flink-job/values.yaml [hunk 3]
```diff
@@ -127,12 +130,17 @@ job:
...
+  # -- (bool) Should take savepoint before upgrading the job (TODO do we want to default to true?)
+  takeSavepointOnUpgrade: false
+  savepointGeneration:
```
Comment (diranged) [inline charts/flink-job/values.yaml:141]: Please add docstrings

File: charts/flink-job/values.yaml [hunk 4]
```diff
@@ -9,6 +9,9 @@ image:
   # -- (String) The Flink image tag
   tag: 1.13.1
 
+# -- batchSchedulerName, see flink-on-k8s-operator docs.
```
Comment (diranged) [inline charts/flink-job/values.yaml:12]: Include a field type here? string etc?.. maybe put a link to the docs?

File: charts/flink-job/values.yaml (hunk 2 above)
Comment (pjthepooh) [reply charts/flink-job/values.yaml:139]: added

File: charts/flink-job/values.yaml (hunk 3 above)
Comment (pjthepooh) [reply charts/flink-job/values.yaml:143]: added

**Pull request merged**